core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -634,6 +634,13 @@ public Result visit(Sort e) {
     }
     Result x = visitChild(0, e.getInput());
     Builder builder = x.builder(e, Clause.ORDER_BY);
+    if (builder.select.getSelectList() == null) {
```

Comment:
Why this change is necessary ?

Reply:
When JdbcSort is used, a regression issue will be trigged: JdbcTest#testSelfJoinDifferentColumns()

RelToSqlConverter visit this Sort and then produce a SQL "SELECT * from (t JOIN t0 ...) order by ...)". However t and t0 include one same name field "last_name"

Before SQL:
select e1."full_name"
  from "foodmart"."employee" as e1
  join "foodmart"."employee" as e2 on e1."first_name" = e2."last_name"
order by e1."last_name" limit 3

REL:
JdbcToEnumerableConverter
  JdbcProject(full_name=[$1], last_name=[$3])
    JdbcSort(sort0=[$3], dir0=[ASC], fetch=[3])
      JdbcJoin(condition=[=($2, $0)], joinType=[inner])
        JdbcProject(last_name=[$3])
          JdbcTableScan(table=[[foodmart, employee]])
        JdbcProject(full_name=[$1], first_name=[$2], last_name=[$3])
          JdbcTableScan(table=[[foodmart, employee]])

After SQL:
SELECT "t2"."full_name", "t2"."last_name0" AS "last_name"
FROM (SELECT *
FROM (SELECT "last_name"
FROM "foodmart"."employee") AS "t"
INNER JOIN (SELECT "full_name", "first_name", "last_name"
FROM "foodmart"."employee") AS "t0" ON "t"."last_name" = "t0"."first_name"
ORDER BY "t0"."last_name" NULLS LAST
LIMIT 3) AS "t2"

Reply:
Without the  change in RelToSqlConverter.java, I still passed  the unchanged `testMatchRecognizePatternExpression12`

Reply:
This change is for `JdbcTest#testSelfJoinDifferentColumns()` mainly, `testMatchRecognizePatternExpression12` is affected by the side effects, so I have to change it

Reply:
Thanks for explanation ~
Since it's very common to execute sql like `select * from XXX order by col`, shall we wrap the change within a check like below:
```
    if (stack.getLast().r != e) {
      ...
    }
```
Thus we can save the change in`RelToSqlConverterTest.java` and keep the converted Sql String simplier.

Reply:
Thank you so much for your help ~ :D

Reply:
Just curious, shouldn't we generate `SELECT "t2"."full_name", "t2"."last_name0" AS "last_name"` instead of `select *` ?

Why we need a decision `if (stack.getLast().r != e)` it looks weird.

Also, the generated sql outputs one more column than the original, i think it is because we generate the sql using the RelRoot.rel directly. We better fix this.

Reply:
I have updated the description of regission issue again, please help review :D

`if (stack.getLast().r != e)` can make a decision whether currently Sort rel is the outermost rel.
If it is, the select list of generated SQL can be "select *", It is more simplier.
If not, the select list of generated SQL shoud contain explicit column name.

`CalcitePrepareImpl#implement(RelRoot root)` generate SQL by root.rel，but the correctly select field be controled by `root.fields`.

Reply:
Hi Danny ~
I gave the suggestion to keep `select *` when `Sort` is the outmost rel, which I think is much simpler.
I'm ok if you think that Calcite should expand the select list for `Sort`.

Reply:
Thanks @jinxing64 , after some research i agree that maybe fixing the sort project list is the easiest way to go, but i would suggest to use `stack.size() == 1` to decide if this sort is a root node instead of `stack.getLast().r != e`.

Reply:
Thank you guys, I have updated it.